cisco_secure_web_appliance/assets/logs/cisco-secure-web-appliance.yaml

```diff
@@ -0,0 +1,438 @@
+    path: network.destination.geoip.continent.code
+    source: log
+  - groups:
+      - Geoip
+    name: Destination Continent Name
+    path: network.destination.geoip.continent.name
+    source: log
+  - groups:
+      - Geoip
+    name: Destination Country ISO Code
+    path: network.destination.geoip.country.iso_code
+    source: log
+  - groups:
+      - Geoip
+    name: Destination Country Name
+    path: network.destination.geoip.country.name
+    source: log
+  - groups:
+      - Geoip
+    name: Destination Subdivision ISO Code
+    path: network.destination.geoip.subdivision.iso_code
+    source: log
+  - groups:
+      - Geoip
+    name: Destination Subdivision Name
+    path: network.destination.geoip.subdivision.name
+    source: log
+  - groups:
+      - Web Access
+    name: Destination IP
+    path: network.destination.ip
+    source: log
+  - groups:
+      - Web Access
+    name: Destination Port
+    path: network.destination.port
+    source: log
+  - groups:
+      - User
+    name: User Name
+    path: usr.name
+    source: log
+pipeline:
+  type: pipeline
+  name: Cisco Secure Web Appliance
+  enabled: true
+  filter:
+    query: "source:cisco-secure-web-appliance"
+  processors:
+    - type: pipeline
+      name: Processing of Access Logs
+      enabled: true
+      filter:
+        query: "service:access_logs "
+      processors:
+        - type: grok-parser
+          name: "Parse access logs"
+          enabled: true
+          source: message
+          samples:
+            - '<38>Sep 23 11:14:10 Crest.CSWA accesslogs: Info: 1727070241.436
+              866 10.10.10.10 TCP_MISS/200 4394 CONNECT
+              tunnel://v10.events.data.microsoft.com:443/ -
+              DIRECT/v10.events.data.microsoft.com application/octet-stream
+              DEFAULT_CASE_12-DefaultGroup-match_network-NONE-NONE-NONE-DefaultGroup-NONE
+              <"IW_comp",9.0,1,"-",-,-,-,1,"-",-,-,-,"-",1,-,"-","-",-,-,"IW_comp",-,"-","Computers
+              and Internet","-","Microsoft Dynamics CRM","Enterprise
+              Applications","Encrypted","-",40.59,0,-,"-","-",1,"-",-,-,"-","-",-,1,"-",-,->
+              - -'
+            - 1726663202.810 4 10.10.10.10 TCP_DENIED/403 0 CONNECT
+              tunnel://FORCEPOINTDLP:443/ - NONE/- -
+              BLOCK_WEBCAT_12-DefaultGroup-match_network-NONE-NONE-NONE-NONE-NONE
+              <"nc",ns,1,"-",-,-,-,-,"-",-,-,-,"-",-,-,"-","-",-,-,"nc",-,"-","-","-","Unknown","Unknown","-","-",0.00,0,-,"-","-",-,"-",-,-,"-","-",-,-,"-",-,->
+              - -
+            - '<38>Sep 24 16:22:24 Crest.CSWA accesslogs: Info: 1727175142.338
+              98 10.10.10.10 TCP_DENIED_SSL/403 0 GET
+              https://www.virustotal.com:443/favicon.ico - NONE/- -
+              BLOCK_CUSTOMCAT_12-DefaultGroup-10.50.6.5-NONE-NONE-NONE-NONE-NONE
+              <"C_viru",-,-,"-",-,-,-,-,"-",-,-,-,"-",-,-,"-","-",-,-,"-",-,"-","-","-","-","-","-","-",0.00,0,-,"-","-",-,"-",-,-,"-","-",-,-,"-",-,->
```

Comment:
This pipeline seems to make the assumption that these 44 attributes will always arrive in this exact order in a comma-separated list, and that for example all logs from the Cisco Web Appliance will have a "YouTube URL" attribute at the end of the list. What is enforcing this format in the web appliance?

Reply:
Yes, the integration supports the default (squid) log style for access logs and the same is also documented in the integration configuration steps.

Reply:
@ankitarajput-crest do you have a link to the cisco documentation that says this is the format?

Reply:
@obi11235 Here is the [link](https://www.cisco.com/c/en/us/td/docs/security/wsa/wsa-14-5/user-guide/wsa-userguide-14-5/b_WSA_UserGuide_11_7_chapter_010101.html#con_1633384) to Cisco Secure Web Appliance documentation which provides the example log. Configuring the default log format option in the Cisco Secure Web Appliance Console will generate and forward the logs in exactly the same format as mentioned above and the same has been successfully validated in our lab environment as well.